test/unit/controllers/declaration/SupervisingCustomsOfficeControllerSpec.scala

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.controllers.declaration
+
+import controllers.declaration.SupervisingCustomsOfficeController
+import helpers.views.declaration.WarehouseIdentificationMessages
+import models.{DeclarationType, Mode}
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito
+import org.mockito.Mockito.{verify, when}
+import org.scalatest.{BeforeAndAfterEach, OptionValues}
+import play.api.libs.json.Json
+import play.api.test.Helpers._
+import play.twirl.api.HtmlFormat
+import unit.base.ControllerSpec
+import views.html.declaration.supervising_customs_office
+
+class SupervisingCustomsOfficeControllerSpec extends ControllerSpec with BeforeAndAfterEach with WarehouseIdentificationMessages with OptionValues {
+
+  val supervisingCustomsOfficeTemplate: supervising_customs_office = mock[supervising_customs_office]
```

Comment:
Inside tests specially for small scope like this there is no value of putting return type for `vals` (for `def` it's always good to add return type). Moreover I don't see value of using word `Template` in this name. It's just make things longer

Reply:
When i remove the return type I get a Type Mismatch error `Required: String, Found: supervising_customs_office` so i've kept it for now unless there's another way around this.
I've removed `Template` from the name

Reply:
When I changed the name and removed the return type I got a Type Mismatch `Required: String, Found: supervising_customs_office`.
I've removed the return type but kept the name the same otherwise I get errors, unless there's another way around this.

Reply:
Yeah, the problem is because you extended `WarehouseIdentificationMessages` and you have a variable `supervisingCustomsOffice` in scope. So return type can be removed, but it's minor comment. Actual it's not a mistake